README.md

```diff
@@ -109,6 +109,43 @@ const catWatch = new ServiceClient({
 });
 ```
 
+## Retry Logic
+
+For application critical requests it can be a good idea to retry failed requests to the responsible services.
+
+Occasionaly target server can have high latency for a short period of time, or in the case of a stack of servers, one server can be having issues
+and retrying the request will allow perron to attempt to access one of the other servers that currently aren't facing issues.
+
+By default `perron` has retry logic implemented, but configured to perform 0 retries. Internally `perron` uses [node-retry](https://github.com/tim-kos/node-retry) to handle the retry logic
+and configuration. All of the existing options provided by `node-retry` can be passed via configuration options through `perron`.
+
+There is a transient error check function. This can be defined in any way by the consumer and is used in the try logic to determine whether to 
+attempt the retries or not depending on the type of error. If the function returns true and the number of retries hasn't been exceeded, the request can be retried.
+
+There is also an onRetry function which can be defined by the user of `perron`. This function is called every time a retry request will be triggered.
+It is provided the currentAttempt, as well as the error that is causing the retry.
```

Comment:
It would be nice to refer to what `currentAttempt` means. Is it 0 or 1 for the first retry?

Reply:
The first time onRetry is triggered it is 1, I'll update the readme with more info.